packages/cpp-debug/src/browser/cpp-memory-widget.tsx

```diff
@@ -308,37 +373,100 @@ export class MemoryView extends ReactWidget {
 
                 // The ASCII view is always in strictly increasing address order.
                 for (const byte of groupBytes) {
-                    asciiStr += isprint(byte) ? String.fromCharCode(byte) : '.';
+                    asciiStr += typeof byte === 'undefined' ?
+                        ' ' : isprint(byte) ? String.fromCharCode(byte) : '.' ;
                 }
             }
 
-            rows.push([addressStr, rowBytesStr, asciiStr])
+            rows.push([rowAddr, addressStr, rowBytesStr, asciiStr])
         }
 
         return <React.Fragment>
             {
                 rows.map((row, index) =>
-                    <tr className='t-mv-view-row' key={index}>
-                        <td className='t-mv-view-address'>{row[0]}</td>
-                        <td className='t-mv-view-data'>{row[1]}</td>
-                        <td className='t-mv-view-code'>{row[2]}</td>
+                    <tr className={ 't-mv-view-row' + (
+                        // Add a marker to help visual navigation when scrolling
+                        row[0] % (this.bytesPerRow * 8) < this.bytesPerRow ?
+                            ' t-mv-view-row-highlight' : '')
+                    } key={index}>
+                        <td className='t-mv-view-address'>{row[1]}</td>
+                        <td className='t-mv-view-data'>{row[2]}</td>
+                        <td className='t-mv-view-code'>{row[3]}</td>
                     </tr>
                 )
             }
         </React.Fragment>;
     }
 
+    protected onWheelMemoryView = (event: React.WheelEvent) => {
+        if (this.memoryReadResult) {
+            const scroll = event.currentTarget.scrollTop;
+            const canScrollDown = scroll >= (event.currentTarget as any).scrollTopMax;
```

Comment:
Unfortunately `scrollTopMax` has [limited browser support](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTopMax) as only Firefox currently supports it.

Instead we can write the following:
```ts
const canScrollDown = scroll >= 0
```

This is true due to the following rule:
```
An element's scrollTop value is a measurement of the distance from the element's 
top to its topmost visible content. When an element's content does not generate a 
vertical scrollbar, then its scrollTop value is 0.
```

Reply:
Thanks for the heads up on `scrollTopMax`, discovered it was here with the autocomplete while debugging, did not figured it would be a Firefox only non-standard feature...

Needs to be replaced indeed, but what you propose will only have the correct behavior when there is no elevator.

Reply:
Fixed.